### PR TITLE
Remove check for apparmor_parser in AppArmor host validation

### PR DESCRIPTION
**Pull request by saschagrunert**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

The `apparmor_parser` binary is not really required for a system to run
AppArmor from a Kubernetes perspective. How to apply the profile is more
in the responsibility of lower level runtimes like CRI-O and containerd,
which may do the binary check on their own.

This synchronizes the current libcontainer implementation with the
vendored Kubernetes source code and allows distributions to use
AppArmor, even when they do not have the parser available in
`/sbin/apparmor_parser`.

**Which issue(s) this PR fixes**:

Refers to https://github.com/opencontainers/runc/pull/2554

**Special notes for your reviewer**:

/cc @timothysc @pjbgf 
/sig node

**Does this PR introduce a user-facing change?**:

```release-note
Removed `/sbin/apparmor_parser` requirement for the AppArmor host validation.
This allows using AppArmor on distributions which ship the binary in a different path.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Review by ehashman — APPROVED** [review]
/triage accepted
/priority important-longterm
/lgtm

**Comment by saschagrunert** [comment]
@ehashman who from SIG node could approve this one for us?

**Review by pjbgf — APPROVED** [review]
/lgtm

**Comment by saschagrunert** [comment]
PTAL @brendandburns @dchen1107 @lavalamp @smarterclayton

**Comment by mrunalp** [comment]
@saschagrunert We already vendor runc. Any reason why we can't use the function from there directly instead of copying?

**Comment by saschagrunert** [comment]
> @saschagrunert We already vendor runc. Any reason why we can't use the function from there directly instead of copying?

Ah yeah I think we can go down that path. Changed as suggested. :+1:

**Comment by saschagrunert** [comment]
/retest
```
I0129 11:46:39.799] Failure [611.958 seconds]
I0129 11:46:39.800] [BeforeSuite] BeforeSuite 
I0129 11:46:39.801] test/e2e/e2e.go:76
I0129 11:46:39.801] 
I0129 11:46:39.801]   BeforeSuite on Node 1 failed
I0129 11:46:39.801] 
I0129 11:46:39.801]   test/e2e/e2e.go:76
I0129 11:46:39.801] ------------------------------
I0129 11:46:39.802] Jan 29 11:46:39.797: INFO: Running AfterSuite actions on all nodes
```

**Comment by saschagrunert** [comment]
Looks like the pods get stuck in pending, wondering if other PRs face the same issue :thinking: 
/retest

**Comment by ehashman** [comment]
@saschagrunert this looks pretty flaky, I'll see if I can file a bug https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&text=BeforeSuite%20on%20Node%201%20failed

**Comment by saschagrunert** [comment]
Had to rebase this one
/assign @dims 
for approval

**Comment by dims** [comment]
/approve
/lgtm

/assign @thockin @smarterclayton 

Tim, Clayton, quick and easy PR. please take a look and `/approve` (needs /pkg approver)

**Comment by ehashman** [comment]
/hold

/cc @dims 

Is this the same containerd failures we've been tracking in #sig-node or is this related to the PR?

**Comment by dims** [comment]
/test

probably the same ones @ehashman ( cc @adisky @SergeyKanzhelev )

**Comment by SergeyKanzhelev** [comment]
/lgtm
/hold cancel

test failures were unrelated

**Pull request merged**
